src/persist/src/indexed/cache.rs

```diff
@@ -130,22 +127,19 @@ impl<B: BlobRead> BlobCache<B> {
     /// thread if it's not already in the cache.
     pub fn get_unsealed_batch_async(&self, key: &str) -> PFuture<Arc<BlobUnsealedBatch>> {
         let (tx, rx) = PFuture::new();
-        {
-            // New scope to ensure the cache lock is dropped during the
-            // (expensive) get.
-            let unsealed = match self.unsealed.lock() {
-                Ok(unsealed) => unsealed,
-                Err(err) => {
-                    tx.fill(Err(err.into()));
-                    return rx;
-                }
-            };
-            if let Some(entry) = unsealed.get(key) {
+        match self.cache.get_unsealed(key) {
+            Err(err) => {
+                tx.fill(Err(err));
+                return rx;
+            }
+            Ok(Some(entry)) => {
                 self.metrics.blob_read_cache_hit_count.inc();
-                tx.fill(Ok(Arc::clone(&entry)));
+                tx.fill(Ok(entry));
                 return rx;
             }
-            self.metrics.blob_read_cache_miss_count.inc();
+            Ok(None) => {
+                self.metrics.blob_read_cache_miss_count.inc();
+            }
```

Comment:
nit: put a comment about this case falling through to the fetch below. I read it wrong the first time I scanned this and someone else might too

Reply:
done